main.tf

```diff
@@ -63,9 +85,27 @@ locals {
   existing_kms_instance_guid = var.kms_encryption_enabled == true ? element(split(":", var.existing_kms_instance_crn), length(split(":", var.existing_kms_instance_crn)) - 3) : null
 }
 
+# Create IAM Authorization Policies to allow event notification to access cos
+resource "ibm_iam_authorization_policy" "cos_policy" {
```

Comment:
I do believe we can scope this to the exact COS bucket instead of all instances in the resource group. Take a look at https://github.ibm.com/GoldenEye/issues/issues/8676